root-packages/fio/Makefile.patch

```diff
@@ -0,0 +1,20 @@
+diff -uNr fio-fio-3.37/Makefile fio-fio-3.37.mod/Makefile
+--- fio-fio-3.37/Makefile	2024-03-27 05:13:51.000000000 +0800
++++ fio-fio-3.37.mod/Makefile	2024-07-18 11:55:40.302184569 +0800
+@@ -9,7 +9,7 @@
+ config-host.mak: configure
+ 	@if [ ! -e "$@" ]; then					\
+ 	  echo "Running configure ...";				\
+-	  ./configure;						\
++	  ./configure --prefix=@TERMUX_PREFIX@;						\
```

Comment:
I think it should be possible to use:
```
termux_step_pre_configure() {
	sed -i "s/@VERSION@/${TERMUX_PKG_VERSION}/g" $TERMUX_PKG_SRCDIR/Makefile
}

termux_step_configure() {
	./configure --prefix $PREFIX $TERMUX_PKG_EXTRA_CONFIGURE_ARGS
}
```

instead, we might want to disable/enable some feature in configure step later and it would be convenient to not have to modify the patch for that.

Reply:
> I think it should be possible to use:
> 
> ```
> termux_step_pre_configure() {
> 	sed -i "s/@VERSION@/${TERMUX_PKG_VERSION}/g" $TERMUX_PKG_SRCDIR/Makefile
> }
> 
> termux_step_configure() {
> 	./configure --prefix $PREFIX $TERMUX_PKG_EXTRA_CONFIGURE_ARGS
> }
> ```
> 
> instead, we might want to disable/enable some feature in configure step later and it would be convenient to not have to modify the patch for that.

The reason I wrote it this way at first was because it would exit if an unsupported option was passed in. I have now made [a patch](https://github.com/termux/termux-packages/pull/20886/files#diff-526a19170da671876426434b90394a5719b87164ea909d4d1315aacfd4ce78ee) to fix this. `TERMUX_PKG_EXTRA_CONFIGURE_ARGS` will be passed in when executing `configure`, so there is no need to execute it manually.

https://github.com/termux/termux-packages/actions/runs/10025667140/job/27709098915?pr=20886#step:8:539